src/status_im2/common/contacts/events.cljs

```diff
@@ -0,0 +1,102 @@
+(ns status-im2.common.contacts.events
+  (:require [clojure.string :as string]
+            [utils.re-frame :as rf]
+            [status-im.utils.types :as types]
+            [re-frame.core :as re-frame]
+            [status-im.ethereum.core :as ethereum]
+            [status-im.ethereum.ens :as ens]
+            [status-im.ethereum.stateofus :as stateofus]
+            [status-im.native-module.core :as status]
+            [status-im.utils.db :as utils.db]
+            [status-im.utils.utils :as utils]))
+
+(re-frame/reg-fx
+ :resolve-public-key-from-ens-name
+ (fn [{:keys [chain-id ens-name on-success on-error]}]
+   (ens/pubkey chain-id ens-name on-success on-error)))
+
+(re-frame/reg-fx
+ :decompress-public-key
+ (fn [{:keys [public-key on-success on-error]}]
+   (status/compressed-key->public-key
+    public-key
+    (fn [resp]
+      (let [{:keys [error]} (types/json->clj resp)]
+        (if error ; If response is of the form {:error "foo"} => ERROR
+          (on-error error)
+          (on-success ; ...and replace "fe701" prefix with "0x" prefix
+           (str "0x" (subs resp 5)))))))))
+
+(rf/defn set-new-identity
+  {:events [:new-contact/set-new-identity]}
+  [{:keys [db]} input]
+  (let [input           (utils/safe-trim input)
+        is-public-key?  (utils.db/valid-public-key? input)
+        compressed-key? (and (string/starts-with? input "zQ3")
+                             (< 26 (count input)))
```

Comment:
Looks like we will need this function in different places as we move to new compressed keys format, maybe we can move it to some utils namespace?

Reply:
Yes, I agree. I moved it out as @briansztamfater suggested(thanks brian!), and per @ibrkhalil added more rigor (thank you ibrahem!):

status-desktop is using [this logic for compressed key validation](https://github.com/status-im/status-desktop/blob/865ed32deb47bd4a302968b2b86e1234b548a097/src/app_service/common/conversion.nim#L10), so I implemented the same validation rules.